File: app/utils/time_utils.py
"""
Time-related utility functions
"""
import time


def github_style_duration(past_time, now=None):
    """Convert time difference to GitHub-style duration string"""
    if now is None:
        now = time.time()
    seconds = int(now - past_time)
    if seconds < 60:
        return f"{seconds}s"
    minutes = seconds // 60
    if minutes < 60:
        return f"{minutes}m"
    hours = minutes // 60
    if hours < 24:
        return f"{hours}h"
    days = hours // 24
    if days < 30:
        return f"{days}d"
    months = days // 30
    if months < 12:
        return f"{months}mo"
    years = months // 12
    return f"{years}y"

File: app/static/time_utils.js
// Time utility functions for formatting and displaying time-related values
// [TODO] Move more time-related utility functions here soon (e.g., formatTimestamp, etc.)

// Format time duration from seconds to human-readable string
function formatDuration(seconds) {
  if (seconds === null) return 'Active';

  const days = Math.floor(seconds / 86400);
  const hours = Math.floor((seconds % 86400) / 3600);
  const minutes = Math.floor((seconds % 3600) / 60);

  const parts = [];
  if (days > 0) {
    parts.push(days + 'd');
  }
  if (days > 0 || hours > 0) {
    parts.push(hours + 'h');
  }
  parts.push(minutes + 'm');

  return parts.join(' ');
}

// Export for Node.js testing if in Node environment
if (typeof module !== 'undefined' && module.exports) {
  module.exports = { formatDuration };
}

